UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -318,11 +412,160 @@ private static string GetLine(string filePath)
+        {
+            logger.LogInformation(EventIds.UploadCommitBatchStart.ToEventId(), "Batch commit for BatchId {batchId} and _X-Correlation-ID:{CorrelationId}", batchCommitMetaData.BatchId, correlationId);
+            List<FileDetail> fileDetails = fileSystemHelper.UploadCommitBatch(batchCommitMetaData);
+            var batchCommitModel = new BatchCommitModel()
+            {
+                FileDetails = fileDetails
+            };
+            HttpResponseMessage httpResponse;
+            httpResponse = await fileShareServiceClient.CommitBatchAsync(HttpMethod.Put, fileShareServiceConfig.Value.BaseUrl, batchCommitMetaData.BatchId, batchCommitModel, batchCommitMetaData.AccessToken, correlationId);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                logger.LogInformation(EventIds.UploadCommitBatchCompleted.ToEventId(), "Batch commit for BatchId {batchId} and _X-Correlation-ID:{CorrelationId} completed", batchCommitMetaData.BatchId, correlationId);
+                return true;
+            }
+            else
+            {
+                logger.LogError(EventIds.UploadCommitBatchNonOkResponse.ToEventId(), "Error while commiting batch for BatchId {batchId} and _X-Correlation-ID:{CorrelationId} completed", batchCommitMetaData.BatchId, correlationId);
+                return false;
+            }
+        }
+
+        public async Task<string> GetBatchStatus(BatchStatusMetaData batchStatusMetaData, string correlationId)
+        {
+            logger.LogInformation(EventIds.GetBatchStatusStart.ToEventId(), "Getting batch status for BatchId {batchId} and _X-Correlation-ID:{CorrelationId}", batchStatusMetaData.BatchId, correlationId);
+            ResponseBatchStatusModel responseBatchStatusModel = new ResponseBatchStatusModel();
+            HttpResponseMessage httpResponse;
+            httpResponse = await fileShareServiceClient.GetBatchStatusAsync(HttpMethod.Get, fileShareServiceConfig.Value.BaseUrl, batchStatusMetaData.BatchId, batchStatusMetaData.AccessToken);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                string bodyJson = await httpResponse.Content.ReadAsStringAsync();
+                responseBatchStatusModel = JsonConvert.DeserializeObject<ResponseBatchStatusModel>(bodyJson);
+                logger.LogInformation(EventIds.GetBatchStatusCompleted.ToEventId(), "Getting batch status for BatchId {batchId} and _X-Correlation-ID:{CorrelationId} completed", batchStatusMetaData.BatchId, correlationId);
+                return responseBatchStatusModel.Status;
+            }
+            else
+            {
+                logger.LogError(EventIds.GetBatchStatusNonOkResponse.ToEventId(), "Error while getting batch status for BatchId {batchId} and _X-Correlation-ID:{CorrelationId} completed", batchStatusMetaData.BatchId, correlationId);
+                return responseBatchStatusModel.Status;
```

Comment:
What value will responseBatchStatusModel.Status return?

Is that what we want or would we be better returning a specific response? (e.g. Failed)

Reply:
Yes, we are returning status which we get from file share service and based on that status we set isUploadZipFile as true or false. In case of committed batch status, we set isUploadZipFile = true else false.